embc-app/ClientApp/src/app/volunteer/components/valid-from-to/valid-from-to.component.ts

```diff
@@ -1,19 +1,157 @@
-import { Component, OnInit, Input } from '@angular/core';
+import { Component, EventEmitter, Input, Output, OnInit } from '@angular/core';
 import range from 'lodash/range';
+import { ReferralDate, ReferralDateForm, YearMonthDay, HourMinute } from 'src/app/core/models/referral-date';
+import * as moment from 'moment';
 
 @Component({
   selector: 'app-valid-from-to',
   templateUrl: './valid-from-to.component.html',
   styleUrls: ['./valid-from-to.component.scss']
 })
 export class ValidFromToComponent implements OnInit {
-  @Input() validFrom: Date = null;
-  @Input() validTo: Date = null;
+  @Input() editMode: boolean; // unimplemented editmode false is read only
```

Comment:
What's the plan to change this to a maker component (ie, read-only) mode? I know the parent component (eg, incidentals) already hard-displays from and to dates, but I think that should be done by THIS component instead. Thoughts?

Reply:
I had discussed this briefly with Alejandro. I think we ideally want a component that can just be a date and time picker. Sub-sub-component would be useful for other places.

Reply:
there is now a read-only mode